iree/base/status_test.cc

```diff
@@ -23,43 +23,60 @@ namespace {
 using ::iree::testing::status::StatusIs;
 using ::testing::HasSubstr;
 
+#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
+#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
+  EXPECT_THAT(status.ToString(), HasSubstr(message_substr))
+#else
+#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
+  EXPECT_THAT(status.ToString(), HasSubstr(StatusCodeToString(status.code())));
+#endif
+
+#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
+#define CHECK_STREAM_SUBSTR(status, os, message_substr) \
+  EXPECT_THAT(os.str(), HasSubstr(message_substr));
+#else
+#define CHECK_STREAM_SUBSTR(status, os, message_substr) \
+  EXPECT_THAT(os.str(), HasSubstr(StatusCodeToString(status.code())));
+#endif
+
```

Comment:
Either always check the code:

```suggestion
#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
#define CHECK_STATUS_MESSAGE(status, code, message_substr) \
  EXPECT_THAT(status.ToString(), HasSubstr(StatusCodeToString(code)));
  EXPECT_THAT(status.ToString(), HasSubstr(message_substr))
#else
#define CHECK_STATUS_MESSAGE(status, code, message_substr) \
  EXPECT_THAT(status.ToString(), HasSubstr(StatusCodeToString(status.code())));
#endif

#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
#define CHECK_STREAM_MESSAGE(os, code, message_substr) \
  EXPECT_THAT(os.str(), HasSubstr(StatusCodeToString(code)));
  EXPECT_THAT(os.str(), HasSubstr(message_substr));
#else
#define CHECK_STREAM_MESSAGE(os, code, message_substr) \
  EXPECT_THAT(os.str(), HasSubstr(StatusCodeToString(status.code())));
#endif
```

Or never do:

```suggestion
#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
  EXPECT_THAT(status.ToString(), HasSubstr(message_substr))
#else
#define CHECK_STATUS_MESSAGE_SUBSTR(status, message_substr) \
  (void) status; (void) message_substr;
#endif

#if (IREE_STATUS_FEATURES & IREE_STATUS_FEATURE_ANNOTATIONS) != 0
#define CHECK_STREAM_SUBSTR(os, message_substr) \
  EXPECT_THAT(os.str(), HasSubstr(message_substr));
#else
#define CHECK_STREAM_SUBSTR(os, message_substr) \
  (void) os; (void) message_substr;
#endif
```

Reply:
Reworked to always check the code.